tests/Models/CompanyJournal.php

```diff
@@ -17,6 +18,21 @@
 class CompanyJournal extends Model
 {
 	use AccountingJournal;
+
+    public $incrementing = false;
+
+    protected static function boot()
+    {
+        parent::boot();
+        static::creating(function ($model) {
+            /**
+             * @var \Illuminate\Database\Eloquent\Model $model
+             */
+            if (!$model->getKey()) {
+                $model->{$model->getKeyName()} = Str::uuid()->toString();
+            }
+        });
```

Comment:
this could be a tiny `trait` for all models. this code block repeats itself more than +3 times.